### ISSUE TITLE
Issue with referencing from one resource to another in same ResourceDictionary

### ISSUE THREAD

**Issue by lsoft** [AvaloniaUI/Avalonia#3323]
Avalonia 0.8.3.

I cannot make a reference from one resource to another via StaticResource even in the same ResourceDictionary. It fails with:

> Avalonia.Markup.Xaml.XamlLoadException: 'Error loading xaml at avares://repro/Visual.xaml: An error occured getting provided value'
> KeyNotFoundException: Static resource 'Red' not found.

In case of change StaticResource to DynamicResource, application starts without exceptions but 'dynamic' reference does not works.

Visual.xaml
```
<ResourceDictionary
    xmlns="https://github.com/avaloniaui"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    >


    <Color x:Key="Red">#FFE21A1A</Color>
    <SolidColorBrush x:Key="RedBrush" Color="{StaticResource ResourceKey=Red}"  />

</ResourceDictionary>
```

MainWindow.xaml
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:s="clr-namespace:System;assembly=mscorlib"
        xmlns:markup="clr-namespace:Portable.Xaml.Markup;assembly=Avalonia.Markup.Xaml"
        xmlns:viewModels="clr-namespace:repro.ViewModels"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="repro.Views.MainWindow"
        Icon="/Assets/avalonia-logo.ico"
        Width="800" Height="600"
        >

    <Design.DataContext>
        <viewModels:MainWindowViewModel/>
    </Design.DataContext>

    <Window.Resources>

      <ResourceDictionary>
        <ResourceDictionary.MergedDictionaries>
          <ResourceInclude Source='/Visual.xaml'/>
        </ResourceDictionary.MergedDictionaries>

      </ResourceDictionary>

    </Window.Resources>

      <Button
          Content="should have red background"
          Margin="40"
          Background="{DynamicResource ResourceKey=RedBrush}"
           />

</Window>
```

Please make note, if I remove RedBrush from Visual.xaml and modify Window.Resources node to

```
<Window.Resources>

      <ResourceDictionary>
        <ResourceDictionary.MergedDictionaries>
          <ResourceInclude Source='/Visual.xaml'/>
        </ResourceDictionary.MergedDictionaries>
        <SolidColorBrush x:Key="RedBrush" Color="{DynamicResource ResourceKey=Red}"  />
      </ResourceDictionary>

</Window.Resources>
```
everything will works fine.

[repro.zip](https://github.com/AvaloniaUI/Avalonia/files/3921745/repro.zip)

**Comment by grokys** [comment]
Confirmed this is a problem: `StaticResource` doesn't work in resource dictionaries. I think we've not noticed this before because top-level `ResourceDictionary`s are very common in Avalonia. Working on a fix now.

**Comment by lsoft** [comment]
Please make note - DynamicResource does not work too with no error message :)

**Comment by grokys** [comment]
Hopefully fixed in https://github.com/AvaloniaUI/Avalonia/pull/3327 if you want to try it.

**Comment by lsoft** [comment]
thanks a lot! could I ask to port that fix back to 0.8 and to release new nuget package with the fix? I can't move to 0.9 because of another issue with ControlTemplate...

**Comment by grokys** [comment]
As we discussed in gitter, the `ControlTemplate` issue was fixed in #3332.
